cpp/src/slice2cpp/Gen.cpp

```diff
@@ -3013,8 +3013,8 @@ Slice::Gen::InterfaceVisitor::visitOperation(const OperationPtr& p)
         if(p->hasMarshaledResult())
         {
             string resultName = marshaledResultStructName(name);
-            params.push_back("::std::function<void(const " + resultName + "&)> " + responsecbParam);
-            args.push_back("[incomingPtr](const " + resultName + "& marshaledResult) { incomingPtr->response(marshaledResult); }");
+            params.push_back("::std::function<void(" + resultName + ")> " + responsecbParam);
```

Comment:
Changed the AMD mapping to take the typed marshaled result struct _by value_. Since this struct has no copy ctor, you need to give this function an xvalue (that's what we do in all the tests) or move-construct from some other instance.